homework/index.js

```diff
@@ -29,22 +33,123 @@
-    createAndAppend('li', ul, { text: repo.name });
+    ul.innerHTML = '';
+    const table = createAndAppend('table', ul, { class: 'table' });
+    const tr1 = addTableRow(table, 'Repository:', '');
+    createAndAppend('a', tr1.lastChild, {
+      href: repo.html_url,
+      text: repo.name,
+    });
+    addTableRow(table, 'Description:', repo.description);
+    addTableRow(table, 'Fork: ', repo.forks);
+    addTableRow(table, 'Updated:', changeDateTimeFormat(repo.updated_at));
+  }
+
+  function renderContributorsDetails(contributorsUrl, ul, root) {
+    ul.innerHTML = '';
+    fetchJSON(contributorsUrl)
+      .catch(err => {
+        createAndAppend('div', root, {
+          text: err.message,
+          class: 'alert-error',
+        });
+      })
+      .then(contributors => {
+        contributors.forEach(contributor => {
+          const contributorLi = createAndAppend('li', ul, {
+            class: 'contributor-list',
+          });
+          createAndAppend('img', contributorLi, {
+            src: contributor.avatar_url,
+            alt: contributor.login,
+            class: 'contributor-image ',
+          });
+          createAndAppend('a', contributorLi, {
+            href: contributor.html_url,
+            text: contributor.login,
+            target: '_blank',
+            class: 'contributor-link ',
+          });
+          createAndAppend('div', contributorLi, {
+            href: contributor.html_url,
+            text: contributor.contributions,
+            class: 'contributor-div ',
+          });
+        });
+      });
+  }
+
+  function bridgeSections(repo, repoList, contributorList) {
+    renderRepoDetails(repo, repoList);
+    renderContributorsDetails(repo.contributors_url, contributorList);
+  }
+
+  function sortAlpha(a, b) {
+    return a.name.localeCompare(b.name);
   }
 
   function main(url) {
-    fetchJSON(url, (err, repos) => {
-      const root = document.getElementById('root');
-      if (err) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root, {
+      class: 'header',
+    });
+    createAndAppend('span', header, {
+      text: 'HYF Repositories',
+      class: 'header-text',
+    });
+    const select = createAndAppend('select', header, { class: 'select-bar' });
+    const mainContainer = createAndAppend('main', root);
+    const repoSection = createAndAppend('section', mainContainer, {
+      class: 'repo-container',
+    });
+    const repoUl = createAndAppend('ul', repoSection);
+    const contributorSection = createAndAppend('section', mainContainer, {
+      class: 'contributors-container',
+    });
+    createAndAppend('h3', contributorSection, {
+      text: 'Contributions',
+      class: 'contributor-title',
+    });
+
+    const contributorUl = createAndAppend('ul', contributorSection);
+
+    fetchJSON(url)
+      .then(repos => {
+        sortedRepos = repos.sort(sortAlpha);
+        sortedRepos.forEach(repo => {
```

Comment:
1. You don't need this global variable `sortedRepos`. You can just chain the `.sort` and `.forEach` like shown below.
2. In the assignment is was hinted to use the index value of the array element for the `value` attribute. This will simplify your code, because you do not need to 'find' the selected repository. You can access it directly by index.
3. You can also the event handler inside the `.then()` method, eliminating the need to define a global variable.

```js
fetchJSON(url)
  .then(repos => {
    repos.sort(sortAlpha).forEach((repo, index) => {
      createAndAppend('option', select, {
        text: repo.name,
        value: index,
      });
    });
    bridgeSections(repos[0], repoUl, contributorUl);
    select.addEventListener('change', event => {
      const selectedRepo = repos[event.target.value];
      bridgeSections(selectedRepo, repoUl, contributorUl);
    });
  ...
```